specifyweb/frontend/js_src/lib/components/Notifications/hooks.tsx

```diff
@@ -29,6 +29,9 @@ export function useNotificationsFetch({
   const [notifications, setNotifications] = React.useState<
     RA<GenericNotification> | undefined
   >(undefined);
+
+  const addSinceToNotificationRef = React.useRef(true);
+
   React.useEffect(() => {
     let pullInterval = INITIAL_INTERVAL;
     let lastFetchedTimestamp: Date | undefined;
```

Comment:
replace this with `const lastFetchDateRef = React.useRef<undefined | Date>(undefined);` and put it outside the useEffect.

and actually, then you don't need addSinceToNotificationRef at all, right? if lastFetchDateRef.current is undefined, then you know that you are in a first fetch, and so "since" needs to be added to the URL
will simplify your code